### DISCUSSION TITLE
Issue while upgrading go.opentelemetry.io/otel/metric from v0.31.0 to v0.37.0

### DISCUSSION THREAD

**Discussion by anil181** [open-telemetry/opentelemetry-go#4022] [Q&A] [answered]
As per the documentation https://github.com/open-telemetry/opentelemetry-go/releases/tag/sdk/metric/v0.32.0  
The packages  go.opentelemetry.io/otel/sdk/metric/selector/simple,  go.opentelemetry.io/otel/sdk/metric/controller/basic (controller), go.opentelemetry.io/otel/sdk/metric/processor/basic (processor)are removed.

As per the existing code a pusher is being created which then periodically collects and exports metrics with the configured interval
```
pusher := controller.New(
		processor.NewFactory(
			simple.NewWithInexpensiveDistribution(),
			metricExp,
		),
		controller.WithExporter(metricExp),
		controller.WithCollectPeriod(2*time.Second),
	)
```
I could not find a proper documentation on how to handle these breaking changes? Could some one guide me to any documentation or examples on handling these changes ? 

Thanks & Reagrds,
Anil

**Comment by pellared** [comment, marked as answer]
AFAIK there is no "migration documentation" other than: https://github.com/open-telemetry/opentelemetry-go/blob/main/CHANGELOG.md

> The metric SDK in `go.opentelemetry.io/otel/sdk/metric` is completely refactored to comply with the OpenTelemetry specification. Please see the package documentation for how the new SDK is initialized and configured. 

See: https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric

**Reply by anil181** [reply]
Thank you Robert, i read but for few updates there is no help or examples.

In the code we have references to some interfaces which are removed but there is not help in the [CHANGELOG.md](https://github.com/open-telemetry/opentelemetry-go/blob/main/CHANGELOG.md)
eg:
otlpmetricgrpc.NewClient
otlpmetric.Client
*otlpmetric.Exporter
*otlpmetricgrpc.Exporter

Could you please suggest or give any pointers on how to replace these?

**Reply by pellared** [reply]
See the following:

- https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc
- https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp
